topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -914,18 +921,31 @@ public void onMaxAttemptsExceeded(MaxAttemptsExceededException e) {
 
         public Reconciliation(String name) {
             this.name = name;
-            this.reconciliationTimerSample = Timer.start(metrics.meterRegistry());
-            reconciliationsCounter.increment();
+            if (isEventWatched()) {
+                LOGGER.debug("Metric {} triggered", this.name);
+                this.reconciliationTimerSample = Timer.start(metrics.meterRegistry());
+                reconciliationsCounter.increment();
+            }
         }
 
         public void failed() {
-            reconciliationTimerSample.stop(reconciliationsTimer);
-            failedReconciliationsCounter.increment();
+            if (isEventWatched()) {
+                LOGGER.debug("failed reconciliation {}", name);
+                reconciliationTimerSample.stop(reconciliationsTimer);
+                failedReconciliationsCounter.increment();
+            }
         }
 
         public void succeeded() {
-            reconciliationTimerSample.stop(reconciliationsTimer);
-            successfulReconciliationsCounter.increment();
+            if (isEventWatched()) {
+                LOGGER.debug("succeeded reconciliation {}", name);
+                reconciliationTimerSample.stop(reconciliationsTimer);
+                successfulReconciliationsCounter.increment();
+            }
+        }
+
+        private boolean isEventWatched() {
+            return !"onResourceEvent".equals(name) && !"reconcile-from-kafka".equals(name);
```

Comment:
Basing it on names seems fragile. I think it would be better to have a constructor parameter for whether a particular `Reconciliation` should count towards metrics.